controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -58,62 +74,183 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+		// we therefore don't need to return error from the reconciler in the case the TC is not found.
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	var reason string
+	resultCondition := clusterCondition
+
+	if clusterCondition == corev1.ConditionTrue {
+		collectUsage := r.GetUsageFunc
+		if collectUsage == nil {
+			collectUsage = collectConsumedCapacity
 		}
-		reportedError = fmt.Errorf("failed to update the SpaceProvisionerConfig status: %w", err)
+		cc, err := collectUsage(ctx, r.Client, spc.Spec.ToolchainCluster, spc.Namespace)
+		if err != nil {
+			updateReadyCondition(spc, corev1.ConditionUnknown, toolchainv1alpha1.SpaceProvisionerConfigFailedToDetermineCapacityReason, err.Error())
+			return err
+		}
+		spc.Status.ConsumedCapacity = cc
+
+		capacityCondition := r.determineCapacityReadyState(spc)
+		if capacityCondition != corev1.ConditionTrue {
+			reason = toolchainv1alpha1.SpaceProvisionerConfigInsufficientCapacityReason
+		} else {
+			reason = toolchainv1alpha1.SpaceProvisionerConfigValidReason
+		}
+
+		resultCondition = capacityCondition
+	} else {
+		reason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotReadyReason
+	}
+
+	readyCondition := toolchainv1alpha1.Condition{
+		Type:   toolchainv1alpha1.ConditionReady,
+		Status: resultCondition,
+		Reason: reason,
 	}
 
-	return ctrl.Result{}, reportedError
+	spc.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(spc.Status.Conditions, readyCondition)
+
+	return nil
 }
 
-func (r *Reconciler) determineReadyState(ctx context.Context, spc *toolchainv1alpha1.SpaceProvisionerConfig) (toolchainv1alpha1.Condition, error) {
+// Note that this function merely mirrors the usage information found in the ToolchainStatus. This means that it actually may work
+// with slightly stale data because the counter.Counts cache might not have been synced yet. This is ok though because the capacity manager
+// doesn't completely rely on the readiness status of the SPC and will re-evaluate the decision taking into the account the contents of
+// the counter cache and therefore completely "fresh" data.
+func collectConsumedCapacity(ctx context.Context, cl runtimeclient.Client, clusterName string, toolchainStatusNs string) (*toolchainv1alpha1.ConsumedCapacity, error) {
+	status := &toolchainv1alpha1.ToolchainStatus{}
+	if err := cl.Get(ctx, types.NamespacedName{Namespace: toolchainStatusNs, Name: toolchainconfig.ToolchainStatusName}, status); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("unable to read ToolchainStatus resource: %w", err)
+	}
+
+	for _, m := range status.Status.Members {
+		if m.ClusterName == clusterName {
+			cc := toolchainv1alpha1.ConsumedCapacity{}
+			cc.MemoryUsagePercentPerNodeRole = m.MemberStatus.ResourceUsage.MemoryUsagePerNodeRole
+			cc.SpaceCount = m.SpaceCount
+
+			return &cc, nil
+		}
+	}
+
+	return nil, nil
```

Comment:
This means that there WAS a toolchain status but we didn't find any info about our cluster in it. I assume there could be some corner case where this could happen - for example while adding or removing a member cluster. Returning no error an nil here is I think the right thing to do. The readiness status of the SPC will be `Unknown` and therefore will not allow provisioning until the ToolchainStatus does contain the capacity info and we can re-reconcile the SPC.

Reply:
I see, thanks for explaining that. Let's keep it as is then 👍